hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java

```diff
@@ -188,6 +188,26 @@ private void associateAccounts(
         for (final var customFee : requireCollectorAutoAssociation) {
             // This should exist as it is validated in validateSemantics
             final var collector = accountStore.get(customFee.feeCollectorAccountIdOrThrow());
+            // TODO: probably this is not the fix but I leave it here to start a discussion.
```

Comment:
I'm open for discussions on that one

Reply:
I think what you did is correct. But we should use similar logic like mono-service using set of accountIds in a set like this ?

Reply:
We can but we have a different structure than the mono. As far as I can understand to have the same effect we need to:
1. Create a new Set<AccountID> associatedSoFar variable in `associateAccounts`
2. Pass it to `createAndLinkTokenRels` to add the linked _accountId_
3. Pass it to `validateAssociation` to verify that the current _accountId_ has not already been linked.

It seems a bit cumbersome. But if you think that it adds additional value let me know and I will implement it 👍

Reply:
I agree, that does seem cumbersome. I'm not sure if this code is equivalent to that code or not; as long as it has the same end result I would think this is OK, but not sure. @Neeharika-Sompalli?

Reply:
yeah that should be fine.